src/pocketmine/plugin/PluginManager.php

```diff
@@ -274,6 +273,14 @@ public function loadPlugins(string $directory, array $newLoaders = null){
 						continue;
 					}
 
+					if(!empty($this->compatibleOperatingSystems) and !in_array(Utils::getOS(), $this->compatibleOperatingSystems)) {
```

Comment:
bad copy pasta

Reply:
please avoid usage of empty(), it includes isset() logic, which is concealing a bug in this code.